### PR TITLE
emulate chosen compiler in integer literal promotion

### PR DESCRIPTION
**Pull request by wrongnull**
Closes https://github.com/Vexu/arocc/issues/690

### CODE REVIEW

File: src/aro/Diagnostics/messages.def [hunk 1]
```diff
@@ -1294,11 +1294,15 @@ invalid_utf8
...
-implicitly_unsigned_literal
+implicitly_unsigned_literal_clang
     .msg = "integer literal is too large to be represented in a signed integer type, interpreting as unsigned"
     .opt = W("implicitly-unsigned-literal")
     .kind = .warning
 
+implicitly_unsigned_literal_gcc
+    .msg = "integer constant is so large that it is unsigned"
+    .kind = .warning
+
```
Comment (Vexu) [inline src/aro/Diagnostics/messages.def:1305]: I don't think we need to emulate differently worded diagnostics too.

File: src/aro/Parser.zig [hunk 2]
```diff
@@ -8448,7 +8448,7 @@ fn fixedSizeInt(p: *Parser, base: u8, buf: []const u8, suffix: NumberSuffix, tok
         const max_int = try Value.maxInt(res.ty, p.comp);
         if (interned_val.compare(.lte, max_int, p.comp)) break;
     } else {
-        res.ty = .{ .specifier = .ulong_long };
+        res.ty = .{ .specifier = if (p.comp.langopts.emulate == .gcc) .int128 else .ulong_long };
```
Comment (Vexu) [inline src/aro/Parser.zig:8451]: This needs to check that the target supports `__int128` with `target_utils.hasFloat128`.

Comment (rurban) [reply src/aro/Parser.zig:8451]: hasFloat128??

Comment (Vexu) [reply src/aro/Parser.zig:8451]: Sorry, `target_utils.hasInt128`. It looks like there is a helper for `hasFloat128` in `Compilation`, you could add the same for `hasInt128`.

Comment (wrongnull) [reply src/aro/Parser.zig]: Done

**Comment by wrongnull** [comment]
@Vexu could you review changes now if you have a chance?

**Comment by Vexu** [comment]
Thanks!

**Pull request merged**